keras/src/optimizers/optimizer.py

```diff
@@ -23,5 +24,90 @@ class Optimizer(BackendOptimizer, base_optimizer.BaseOptimizer):
     pass
 
 
+@keras_export("keras.optimizers.VariableUpdater")
+class VariableUpdater:
+    """Allows special handling of variable updates."""
```

Comment:
If this is intended to be public in the API then we should provide an explanation and a usage example in the docstring

Reply:
Great!  Yes, I'd like it to be public so I can inherit from it without needing to import internal keras APIs.

Is this location (within `optimizer.py`) okay, or would you prefer it somewhere else, (e.g. in its own file)?